geode-core/src/main/java/org/apache/geode/management/internal/cli/commands/ListDiskStoreCommand.java

```diff
@@ -0,0 +1,120 @@
+  @ResourceOperation(resource = ResourcePermission.Resource.CLUSTER,
+      operation = ResourcePermission.Operation.READ)
+  public Result listDiskStore() {
+    try {
+      Set<DistributedMember> dataMembers = getNormalMembers(getCache());
+
+      if (dataMembers.isEmpty()) {
+        return ResultBuilder.createInfoResult(CliStrings.NO_CACHING_MEMBERS_FOUND_MESSAGE);
+      }
+
+      return toTabularResult(getDiskStoreListing(dataMembers));
+    } catch (FunctionInvocationTargetException ignore) {
+      return ResultBuilder.createGemFireErrorResult(CliStrings
+          .format(CliStrings.COULD_NOT_EXECUTE_COMMAND_TRY_AGAIN, CliStrings.LIST_DISK_STORE));
+    } catch (VirtualMachineError e) {
+      SystemFailure.initiateFailure(e);
+      throw e;
+    } catch (Throwable t) {
+      SystemFailure.checkFailure();
+      return ResultBuilder.createGemFireErrorResult(
+          String.format(CliStrings.LIST_DISK_STORE__ERROR_MESSAGE, toString(t, isDebugging())));
+    }
+  }
+
+  @SuppressWarnings("unchecked")
+  public List<DiskStoreDetails> getDiskStoreListing(Set<DistributedMember> members) {
+    final Execution membersFunctionExecutor = getMembersFunctionExecutor(members);
+    if (membersFunctionExecutor instanceof AbstractExecution) {
+      ((AbstractExecution) membersFunctionExecutor).setIgnoreDepartedMembers(true);
+    }
+
+    final ResultCollector<?, ?> resultCollector =
+        membersFunctionExecutor.execute(new ListDiskStoresFunction());
+
+    final List<?> results = (List<?>) resultCollector.getResult();
+    final List<DiskStoreDetails> distributedSystemMemberDiskStores =
+        new ArrayList<>(results.size());
+
+    for (final Object result : results) {
+      if (result instanceof Set) { // ignore FunctionInvocationTargetExceptions and other
+        // Exceptions...
+        distributedSystemMemberDiskStores.addAll((Set<DiskStoreDetails>) result);
+      }
+    }
+
+    Collections.sort(distributedSystemMemberDiskStores);
+
+    return distributedSystemMemberDiskStores;
+  }
+
+  private Result toTabularResult(final List<DiskStoreDetails> diskStoreList)
+      throws ResultDataException {
+    if (!diskStoreList.isEmpty()) {
+      final TabularResultData diskStoreData = ResultBuilder.createTabularResultData();
+
+      for (final DiskStoreDetails diskStoreDetails : diskStoreList) {
+        diskStoreData.accumulate("Member Name", diskStoreDetails.getMemberName());
+        diskStoreData.accumulate("Member Id", diskStoreDetails.getMemberId());
+        diskStoreData.accumulate("Disk Store Name", diskStoreDetails.getName());
+        diskStoreData.accumulate("Disk Store ID", diskStoreDetails.getId());
+      }
+
+      return ResultBuilder.buildResult(diskStoreData);
+    } else {
+      return ResultBuilder
+          .createInfoResult(CliStrings.LIST_DISK_STORE__DISK_STORES_NOT_FOUND_MESSAGE);
+    }
+  }
+
+  public Set<DistributedMember> getNormalMembers(final InternalCache cache) {
```

Comment:
Consider pulling this method into the Utils class, it's used here and in ShowMissginDiskStoresCommand. On the other hand you could inline the call to CliUtil.getAllNormalMembers; but since there seems to be some question as to it's purpose I would keep it as a utility method for now.